src/main/kotlin/ink/abb/pogo/scraper/util/directions/RouteProviderEnum.kt

```diff
@@ -0,0 +1,134 @@
+import com.pokegoapi.google.common.geometry.S2LatLng
+import java.util.*
+import java.util.regex.Pattern
+
+
+enum class RouteProviderEnum {
+
+    MAPZEN {
+        override fun createURLString(startLat: Double, startLong: Double, endLat: Double, endLong: Double): String {
+            return "http://valhalla.mapzen.com/route?json={\"locations\":[{\"lat\":$startLat,\"lon\":$startLong},{\"lat\":$endLat,\"lon\":$endLong}],\"costing\":\"pedestrian\",\"directions_options\":{\"narrative\":\"false\"}}"
+        }
+
+        override fun parseRouteResponse(routeParsed: String): ArrayList<S2LatLng> {
+
+            val jsonRoot = ObjectMapper().readTree(routeParsed)
+            val status = jsonRoot.path("trip").path("status").asInt()
+
+            // status 0 == no problem
+            if (status == 0) {
+                val shape = jsonRoot.path("trip").findValue("shape").textValue()
+
+                // Decode the route shape, look at https://mapzen.com/documentation/turn-by-turn/decoding/
+                val precision: Double = 1E6
+                val latlngList = ArrayList<S2LatLng>()
+                var index: Int = 0
+                var lat: Int = 0
+                var lng: Int = 0
+                while (index < shape.length) {
+                    var b: Int
+                    var shift = 0
+                    var result = 0
+                    do {
+                        b = shape[index++].toInt() - 63
+                        result = result or (b and 0x1f shl shift)
+                        shift += 5
+                    } while (b >= 0x20)
+                    val endLat = if (result and 1 != 0) (result shr 1).inv() else result shr 1
+                    lat += endLat
+
+                    shift = 0
+                    result = 0
+                    do {
+                        b = shape[index++].toInt() - 63
+                        result = result or (b and 0x1f shl shift)
+                        shift += 5
+                    } while (b >= 0x20)
+                    val endLong = if (result and 1 != 0) (result shr 1).inv() else result shr 1
+                    lng += endLong
+
+                    latlngList.add(S2LatLng.fromDegrees(lat / precision, lng / precision))
+                }
+                return latlngList // everything is ok
+            }
+            return ArrayList() // can't parse
+        }
+    },
+
+    MOBROUTING {
+        override fun createURLString(startLat: Double, startLong: Double, endLat: Double, endLong: Double): String {
+            return "http://mobrouting.com/api/dev/gosmore.php?flat=$startLat&flon=$startLong&tlat=$endLat&tlon=$endLong&v=foot&fast=1&layer=mapnik"
+        }
+
+        override fun parseRouteResponse(routeParsed: String): ArrayList<S2LatLng> {
+            if (!routeParsed.contains("<distance>0</distance>")) {
+                val matcher = Pattern.compile("(|-)\\d+.\\d+,(|-)\\d+.\\d+").matcher(routeParsed.split("<coordinates>")[1])
```

Comment:
If this external service returns a 522 error (because overload), the routeParsed.split()[1] part will throw an exception and lock the walking completely (see also #1162).